sale_commission_agent_restrict/models/res_partner.py

```diff
@@ -35,3 +35,19 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    def _update_fields_values(self, fields):
+        res = super(ResPartner, self)._update_fields_values(fields)
+        for agent in self.agent_ids:
+            for user_id in agent.user_ids:
+                if (
+                    "agent_ids" in res.keys()
```

Comment:
`"agent_ids" in res` this check should be done before the main loop IMO.
Also, `user_id` is actually a `user`.